mitosheet/mitosheet/parser.py

```diff
@@ -10,12 +10,23 @@
 """
 import datetime
 import re
-from typing import Any, Collection, List, Optional, Set, Tuple, Union
+import warnings
+from typing import Any, List, Optional, Set, Tuple, Union
+
+import pandas as pd
 
 from mitosheet.column_headers import get_column_header_display
 from mitosheet.errors import make_invalid_formula_error
-from mitosheet.transpiler.transpile_utils import column_header_to_transpiled_code
-from mitosheet.types import ColumnHeader
+from mitosheet.sheet_functions.types.utils import is_number_dtype, is_datetime_dtype, is_string_dtype
+from mitosheet.transpiler.transpile_utils import \
+    column_header_to_transpiled_code
+from mitosheet.types import (ColumnHeader, FrontendFormula, ParserMatch, ParserMatchRange,
+                             RowOffset)
+
+
+COLUMN_HEADER_MATCH_TYPE = 'column header match type'
+INDEX_LABEL_MATCH_TYPE = 'index label match type'
```

Comment:
Can you leave a comment for each of these explaining what the match represents? My understanding is that if given the formula =A0 in B1, the parser matches would be:

```
{
                    'type': COLUMN_HEADER_MATCH_TYPE,
                    'match_range': (index_label_start, index_label_start + len(number_chars)),
                    'unparsed':'A',
                    'parsed': df['A'],
                    'row_offset': -1
}
{
                    'type': INDEX_LABEL_MATCH_TYPE,
                    'match_range': (index_label_start, index_label_start + len(number_chars)),
                    'unparsed':'0',
                    'parsed': 0,
                    'row_offset': -1
}

```

Reply:
Close. I updated the comment.

Reply:
You're off actually:
1. the match_range (has to be the range in the formula of the match), 
2. in the parsed column header, it just stays as A -- it would change if it were a number or something for example, and get parsed into an actual int)

Otherwise, is good! I added as a comment. They would also be returned in reverse order.